napari/layers/points/points.py

```diff
@@ -275,6 +279,25 @@ def __init__(
             properties = dataframe_to_properties(properties)
         self._properties = self._validate_properties(properties)
 
+        if default_properties is None:
+            default_properties = {}
+        elif not isinstance(default_properties, dict):
+            default_properties = dataframe_to_properties(default_properties)
+        self._default_properties = default_properties
+
+        if len(data) == 0 and self.properties and not default_properties:
+            raise ValueError(
+                'default_properties must be set for empty Points layers with colors that depend on properties'
```

Comment:
Remember that size and symbol should eventually also be settable by properties, so "colors" isn't right here. Let's say, "properties_choices must be set for empty Points layers with properties".